test/codeu/chat/server/ConversationAccessServerTest.java

```diff
@@ -0,0 +1,535 @@
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ *  Note: The following names of methods in this tester are tentative. They are just  
+ *  placeholders (if you know a better name for a method, tell me!) and will mirror the 
+ *  corresponding methods as we implement them, so let me know when you push a new commit
+ *  that changes the names of what I have listed here. ~ Sarah Abowitz
+ *
+ *  Replaceable parts:
+ *  - codeu.chat.server.Controller.getPermissionMap(ConversationHeader conversation)
+ *  - codeu.chat.server.Controller.updatePermissionMap(Uuid id, HashMap<Uuid, byte> accessMap)
+ *  - codeu.chat.server.Controller.checkMembership(Uuid id, ConversationHeader conversation)
+ *  - ...server.Controller.demoteOwner(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.addOwner(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.ban(byte access, Uuid target, ConversationHeader conversation)
+ *  - ...server.Controller.addMember(byte access, Uuid target, ConversationHeader conversation)
+ *       
+ */
+
+
+package codeu.chat.server;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+
+import static org.junit.Assert.*;
+import org.junit.Test;
+import org.junit.Before;
+
+import codeu.chat.common.BasicController;
+import codeu.chat.common.BasicView;
+import codeu.chat.common.ConversationHeader;
+import codeu.chat.common.Message;
+import codeu.chat.common.User;
+import codeu.chat.util.Uuid;
+
+public final class ConversationAccessServerTest {
+
+  private Model model;
+  private BasicController controller;
+  private BasicView view;
+  private byte creatorByte = 0b111; // Before testing, this must be the equivalent of 111.
+  private byte ownerByte = 0b011; // Before testing, this must be the equivalent of 011.
+  private byte memberByte = 0b001; // Before testing, this must be the equivalent of 001.
+  private byte trollByte = 0b000; // Before testing, this must be the equivalent of 000.
+  
+  @Before
+  public void doBefore() {
+    model = new Model();
+    controller = new Controller(Uuid.NULL, model);
+    view = new View(model);			
+  }
+
+  @Test
+  public void testAddConversation() {
+
+    final User user = controller.newUser("user");
+
+    assertNotNull(
+        "Check that user has a valid reference",
+        user);
+
+    final ConversationHeader conversation = controller.newConversation(
+        "conversation",
+        user.id);
+
+    assertFalse(
+        "Check that conversation has a valid reference",
+        conversation == null);    
+        
+    assertNotNull(
```

Comment:
You should go a step further here and assert that the contents of the permission map that was returned by `controller.getPermissionMap(conversation)` are correct.